orttraining/orttraining/models/gpt2/main.cc

```diff
@@ -227,6 +221,15 @@ Status ParseArguments(int argc, char* argv[], GPT2Parameters& params, OrtParamet
     ORT_RETURN_IF_NOT(params.data_parallel_size > 0, "data_parallel_size must > 0");
     ORT_RETURN_IF_NOT(params.horizontal_parallel_size > 0, "horizontal_parallel_size must > 0");
 
+    int seed = flags["seed"].as<int>();
+    if (params.horizontal_parallel_size > 1 && seed <= 0) {
+      seed = 8211; // Megatron needs a random seed.
```

Comment:
what about utils::GetRandomSeed() instead of 8211?

Reply:
`utils::GetRandomSeed()` by default uses the clock for the seed (until `utils::SetRandomSeed()` is called).  So when running distributed jobs (i.e. multiple processes), each process will have a different seed.

When running with horizontal parallelism (aka Megatron-style model splitting), we are running with multiple processes.  However, we need to ensure all processes use the same seed for Dropout.

So the intent of the code here is to ensure we get a globally agreed upon initial seed, either from the user or some static value (arbitrarily 8211).